d3b_utils/aws_bucket_contents.py

```diff
@@ -0,0 +1,81 @@
+import boto3
+import pandas as pd
+
+
+def fetch_aws_bucket_obj_info(bucket_name, path_to_files):
+    """
+    List the objects of a bucket and the size of each object. Returns a
+    dict. Inspiration from
+    https://alexwlchan.net/2019/07/listing-s3-keys/
+
+    :param bucket_name: The name of the bucket.
+    :param path_to_files: path to the specific folder in the given
+        bucket where the files you want to list are. Exclude leading
+        '/' but include a '/' at the end of the folder. Should look
+        like 'path/to/files/'. This can also be a tuple of paths.
+    """
+
+    client = boto3.client(service_name="s3")
+
+    # Create a paginator. This is needed to page through results when the
+    # number of objects in a bucket is greater than 1000.
+
+    paginator = client.get_paginator("list_objects_v2")
+
+    kwargs = {"Bucket": bucket_name}
+
+    # We can pass the prefix directly to the S3 API.  If the user has passed
+    # a tuple or list of prefixes, we go through them one by one.
+    if isinstance(path_to_files, str):
+        prefixes = (path_to_files,)
+    else:
+        prefixes = path_to_files
+
+    # Build the bucket contents dict
+    bucket_contents = []
+
+    for key_prefix in prefixes:
+        kwargs["Prefix"] = key_prefix
+
+        for page in paginator.paginate(**kwargs):
+            try:
+                contents = page["Contents"]
+            except KeyError:
+                return
+
+            for obj in contents:
+                bucket_contents.append(
+                    {"object_name": obj["Key"], "object_size": obj["Size"]}
+                )
+
+    return bucket_contents
+
+
+def get_bucket_as_df(bucket_name, bucket_path):
```

Comment:
I don't think this function belongs in this module. Separating the file name from the whole path and optionally removing folders should probably be done in fetch_aws_bucket_obj_info. Leave the pandas dependency for the person calling fetch_aws_bucket_obj_info when they decide to put a DataFrame around the result themselves.